server/src/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -852,7 +852,8 @@ public boolean releaseSecondaryIpFromNic(long ipAddressId) {
                 throw new InvalidParameterValueException("Can' remove the ip " + secondaryIp + "is associate with static NAT rule public IP address id " + publicIpVO.getId());
             }
 
-            if (_lbService.isLbRuleMappedToVmGuestIp(secondaryIp)) {
+            List<Long> lbRuleIdList = _firewallDao.listIdByNetworkAndPurposeAndNotRevoked(network.getId(), Purpose.LoadBalancing);
```

Comment:
@ustcweizhou  : As per your comment, do you expect Null and Empty check for list variable "lbRuleIdList " to be moved in LoadBalancerVMMapDao.java class ?